modules/globebrowsing/util/globelabelscomponent.cpp

```diff
@@ -0,0 +1,846 @@
+                    bool enabled = dictionary.value<bool>(
+                        LabelsFadeOutEnabledInfo.identifier
+                    );
+                    _labelsFadeInEnabled.set(enabled);
+                }
+
+                if (dictionary.hasKey(LabelsFadeOutStartingDistanceInfo.identifier)) {
+                    float dist = dictionary.value<float>(
+                        LabelsFadeOutStartingDistanceInfo.identifier
+                    );
+                    _labelsFadeOutDist.set(dist);
+                }
+
+                if (dictionary.hasKey(LabelsMinSizeInfo.identifier)) {
+                    int size = static_cast<int>(
+                        dictionary.value<float>(LabelsMinSizeInfo.identifier)
+                    );
+                    _labelsMinSize.set(size);
+                }
+
+                if (dictionary.hasKey(LabelsMaxSizeInfo.identifier)) {
+                    int size = static_cast<int>(
+                        dictionary.value<float>(LabelsMaxSizeInfo.identifier)
+                    );
+                    _labelsMaxSize.set(size);
+                }
+
+                if (dictionary.hasKey(LabelsDisableCullingEnabledInfo.identifier)) {
+                    bool disabled = dictionary.value<bool>(
+                        LabelsDisableCullingEnabledInfo.identifier
+                    );
+                    _labelsDisableCullingEnabled.set(disabled);
+                }
+
+                if (dictionary.hasKey(LabelsDistanceEPSInfo.identifier)) {
+                    float dist = static_cast<float>(
+                        dictionary.value<double>(LabelsDistanceEPSInfo.identifier)
+                    );
+                    _labelsDistaneEPS.set(dist);
+                }
+
+                if (dictionary.hasKey(LabelAlignmentOptionInfo.identifier)) {
+                    std::string labelAlignment = 
+                        dictionary.value<std::string>(LabelAlignmentOptionInfo.identifier);
+                    if (labelAlignment == "Horizontally") {
+                        _labelAlignmentOption = Horizontally;
+                    }
+                    else {
+                        _labelAlignmentOption = Circularly;
+                    }
+                }
+
+                _font = font;
+                initializeFonts();
+            }
+        }
+    }
+}
+
+bool GlobeLabelsComponent::initializeGL() {
+    return true;
+}
+
+void GlobeLabelsComponent::initializeFonts() {
+    if (_font == nullptr) {
+        _font = openspace::global::fontManager.font(
+            "Mono",
+            static_cast<float>(_labelsFontSize),
+            ghoul::fontrendering::FontManager::Outline::Yes,
+            ghoul::fontrendering::FontManager::LoadGlyphs::No
+        );
+    }
+}
+
+
+bool GlobeLabelsComponent::deinitialize() {
+    return true;
+}
+
+bool GlobeLabelsComponent::isReady() const {
+    return true;
+}
+
+void GlobeLabelsComponent::update() {
```

Comment:
Can the `initializeGL`, `deinitialize`, and `update` methods go away if they don't do anything?